### ISSUE TITLE
Improve the performance of applying groups and users (group_config / user_config)

### ISSUE THREAD

**Issue by tomasalmeida** [Netcentric/accesscontroltool#503]
If a setup has a large number of groups and users (in particular the number of users can be large like > 100,000), the execution time of the AC Tool can get fairly long. If the AC tool is triggered by the install hook that is included in a package, the installation using the package manager might run into a timeout:

```
[INFO] Package uploaded, now installing...
[INFO] ERROR: HTTP call to http://localhost:4503/crx/packmgr/service/console.html/etc/packages/amg/package-1.0.0-SNAPSHOT.zip?cmd=install&recursive=true failed: Read timed out (consider to increase the socket timeout using -Dvault.httpSocketTimeoutSec)
[INFO] ---------------
[INFO] HTTP call failed, try again (1/24) after 5 second(s)...
```

Increasing the timeout can solve the problem, but in some special cases that is not enough. 

One of the bottlenecks is `AuthorizableInstallerServiceImpl.getDeclaredMembersWithoutRegularUsers` which analyses each member of the group linearly. A solution with jdk8+ would be using a parallel stream to speed up this process, using jdk7, I believe we need to handle the threads manually.

**Comment by ghenzler** [comment]
I think the following would be worth a try:
* Run the whole loop of applying each authorizable in parallel https://github.com/Netcentric/accesscontroltool/blob/382c122dac7beb94fd8d0d6d6587c58e43a06194/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/authorizableinstaller/impl/AuthorizableInstallerServiceImpl.java#L110 in parallel (this should be more beneficial than doing this just for `AuthorizableInstallerServiceImpl.getDeclaredMembersWithoutRegularUsers`
* To be able to use Java 1.8 features we can move the build to java 1.8 (while the target should remain 1.7 until we fix #400 @kwin do you see any problem using 1.8 just for the build with target 1.7?)
* to be able to compare the performance between parallel and sequential performance, we should introduce a feature flag `applyAuthorizablesInParllel: true` in section `global_config`

**Comment by tomasalmeida** [comment]
I was trying to set 1.8 as source, but after reading https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-source-and-target.html, it is clearly not possible 
`The pitfall is unintended usage of APIs that only exist in later JREs which would make your code fail at runtime with a linkage error.`

I will rollback the changes to compile jdk 1.8, in order to avoid using not available features in jdk1.7
